util/reggen/uvm_reg_base.sv.tpl

```diff
@@ -37,11 +57,21 @@ ${make_ral_pkg_hdr(dv_base_prefix, [])}
 ${make_ral_pkg_fwd_decls(esc_if_name, rb.type_regs, rb.windows)}
 % for r in rb.all_regs:
 <%
-    reg = r.reg if isinstance(r, MultiRegister) else r
-    assert isinstance(reg, Register)
+    regs = []
+    if isinstance(r, MultiRegister):
+      reg = r.reg
+      if r.dv_compact:
+        reg.fields = r.regs[0].fields
+        regs.append(reg)
+      else:
+        regs.extend(r.regs)
+    else:
+      regs.append(r)
 %>\
 
+  % for reg in regs:
```

Comment:
Do these lines need switching? I think we probably want a blank line between each class.

Reply:
Good catch. fixed. thanks